test/e2e/apimachinery/storage_version.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"time"
+
+	apiserverinternalv1alpha1 "k8s.io/api/apiserverinternal/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/onsi/ginkgo"
+)
+
+const (
+	svName     = "storageversion.e2e.test.foos"
+	idNonExist = "id-non-exist"
+)
+
+var _ = SIGDescribe("StorageVersion resources [Feature:StorageVersionAPI]", func() {
+	f := framework.NewDefaultFramework("storage-version")
+
+	ginkgo.It("storage version with non-existing id should be GC'ed", func() {
+		client := f.ClientSet
+		sv := &apiserverinternalv1alpha1.StorageVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: svName,
+			},
+		}
+		createdSV, err := client.InternalV1alpha1().StorageVersions().Create(context.TODO(), sv, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "creating storage version")
+
+		// update the created sv with server storage version
+		version := "v1"
+		createdSV.Status = apiserverinternalv1alpha1.StorageVersionStatus{
+			StorageVersions: []apiserverinternalv1alpha1.ServerStorageVersion{
+				{
+					APIServerID:       idNonExist,
+					EncodingVersion:   version,
+					DecodableVersions: []string{version},
+				},
+			},
+			CommonEncodingVersion: &version,
+		}
+		_, err = client.InternalV1alpha1().StorageVersions().UpdateStatus(
+			context.TODO(), createdSV, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "updating storage version")
+
+		// wait for sv to be GC'ed
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			_, err := client.InternalV1alpha1().StorageVersions().Get(
+				context.TODO(), createdSV.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			return false, nil
```

Comment:
log something here so it's clear the test found it and is retrying

Reply:
Added logs
```
I0108 20:50:45.580] [It] storage version with non-existing id should be GC'ed
I0108 20:50:45.581]   test/e2e/apimachinery/storage_version.go:40
I0108 20:50:45.655] Jan  8 20:50:45.655: INFO: Waiting for storage version storageversion.e2e.test.foosmgwvb to be garbage collected
I0108 20:50:45.690] [AfterEach] [sig-api-machinery] StorageVersion resources [Feature:StorageVersionAPI]
```